### PR TITLE
disk/CreateVolume: properly set error code

### PR DESCRIPTION
**Pull request by huww98**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

properly set error code for:
* not existing snapshot
* volume already created with the same name but different parameters

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/disk/controllerserver.go [hunk 1]
```diff
@@ -224,6 +226,15 @@ func (cs *controllerServer) CreateVolume(ctx context.Context, req *csi.CreateVol
 
 	diskType, diskID, diskPL, err := createDisk(req.GetName(), snapshotID, requestGB, diskVol, req.Parameters[TenantUserUID])
 	if err != nil {
+		var aliErr *alicloudErr.ServerError
+		if errors.As(err, &aliErr) {
```
Comment (tydra-wang) [inline pkg/disk/controllerserver.go:230]: it will panic because aliErr is nil?

Comment (huww98) [reply pkg/disk/controllerserver.go:230]: `&aliErr` is of type `**alicloudErr.ServerError` and `&aliErr` is not nil. Besides, this is already covered by the csi-sanity test cases.

**Pull request merged**